sdk/monitor/Azure.Monitor.Query/src/MetricsBatchQueryClient.cs

```diff
@@ -0,0 +1,214 @@
+        /// <param name="options">The <see cref="MetricsBatchQueryClientOptions"/> instance to as client configuration.</param>
+        public MetricsBatchQueryClient(TokenCredential credential, MetricsBatchQueryClientOptions options) : this(_defaultEndpoint, credential, options)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/>.
+        /// </summary>
+        /// <param name="endpoint">The data plane service endpoint to use. For example <c>https://metrics.monitor.azure.com/.default</c> for public cloud.</param>
+        /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
+        /// <param name="options">The <see cref="MetricsQueryClientOptions"/> instance to as client configuration.</param>
+        public MetricsBatchQueryClient(Uri endpoint, TokenCredential credential, MetricsBatchQueryClientOptions options = null)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            options ??= new MetricsBatchQueryClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            var scope = "https://metrics.monitor.azure.com/.default";
+            Endpoint = endpoint;
+
+            var pipeline = HttpPipelineBuilder.Build(options,
+                new BearerTokenAuthenticationPolicy(credential, scope));
+
+            _metricBatchClient = new MetricsBatchRestClient(_clientDiagnostics, pipeline, endpoint);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/> for mocking.
+        /// </summary>
+        protected MetricsBatchQueryClient()
+        {
+        }
+
+        /// <summary>
+        /// Gets the endpoint used by the client.
+        /// </summary>
+        public Uri Endpoint { get; }
+
+        /// <summary>
+        /// pass in a list of resource ids, metric names, and a time range to query metrics for those resources.
+        /// </summary>
+        /// <param name="resourceIds"></param>
+        /// <param name="metricNames"></param>
+        /// <param name="metricNamespace"></param>
+        /// <param name="options"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual Response<MetricResultsResponse> Batch(List<string> resourceIds, List<string> metricNames, string metricNamespace, MetricsQueryOptions options = null, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsBatchQueryClient)}.{nameof(Batch)}");
+            scope.Start();
+
+            try
+            {
+                if (resourceIds.Count == 0)
+                {
+                    throw new ArgumentException("Resource IDs can not be empty");
```

Comment:
This sort of argument validation should probably occur before we open the diagnostic scope.